### ISSUE TITLE
Error while saving block in database

### ISSUE THREAD

**Issue by farukterzioglu** [dgarage/NBXplorer#338]
Recently we are getting the exception below;

We have the latest version and this exception is repeating.

```csharp
warn: Explorer:       BTC: Error while saving block in database, retrying in 1 seconds (Invalid argument)
System.IO.IOException: Invalid argument
   at System.IO.FileStream.CheckFileCall(Int64 result, Boolean ignoreNotSupported)
   at System.IO.FileStream.Seek(Int64 offset, SeekOrigin origin)
   at DBTrie.Storage.FileStorage.Read(Int64 offset, Memory`1 output)
   at DBTrie.Storage.CacheStorage.FetchPage(Int32 p)
   at DBTrie.Storage.CacheStorage.Write(Int64 offset, ReadOnlyMemory`1 input)
   at DBTrie.Storage.StorageExtensions.WriteToEnd(IStorage storage, ReadOnlyMemory`1 input)
   at DBTrie.TrieModel.LTrieNode.WriteNewValue(ReadOnlyMemory`1 key, ReadOnlyMemory`1 value)
   at DBTrie.TrieModel.LTrieNode.SetExternalValue(Byte label, ReadOnlyMemory`1 key, ReadOnlyMemory`1 value)
   at DBTrie.TrieModel.LTrieNode.SetValue(Nullable`1 linkLabel, ReadOnlyMemory`1 key, ReadOnlyMemory`1 value)
   at DBTrie.TrieModel.LTrie.SetValue(ReadOnlyMemory`1 key, ReadOnlyMemory`1 value, Boolean replace)
   at DBTrie.Table.Insert(ReadOnlyMemory`1 key, ReadOnlyMemory`1 value, Boolean replace)
   at NBXplorer.Repository.SaveTransactions(DateTimeOffset now, Transaction[] transactions, uint256 blockHash) in /NBXplorer/NBXplorer/Repository.cs:line 793
   at NBXplorer.ExplorerBehavior.SaveMatches(TrackedTransaction[] matches, uint256 blockHash, DateTimeOffset now, Boolean fireEvents) in /NBXplorer/NBXplorer/Expl
orerBehavior.cs:line 230
   at NBXplorer.ExplorerBehavior.SaveMatches(Block block) in /NBXplorer/NBXplorer/ExplorerBehavior.cs:line 186
```

![image](https://user-images.githubusercontent.com/476514/132834123-8aee6841-8f47-4824-9220-15369b1c3ff6.png)

**Comment by farukterzioglu** [comment]
Even I start it with old data, it throws error at the same block 
![image](https://user-images.githubusercontent.com/476514/132839862-f0359c30-845c-4155-9353-1e4ae7f47106.png)

**Comment by farukterzioglu** [comment]
throw in here https://github.com/dgarage/NBXplorer/blob/d466b64836f44714a57e402b9d8ff8ab759b3c2b/NBXplorer/Repository.cs#L793

**Comment by farukterzioglu** [comment]
I tried to rescan, also got this exception on different blocks (e.g. 699767, 699768)

**Comment by NicolasDorier** [comment]
DBTrie, a dependency to NBXPlorer had a critical bug which result in a corrupted install which can't restart. Fixed in v2.2.7.
This impact only big instances.

Thanks for the report!
